### DISCUSSION TITLE
get table names

### DISCUSSION THREAD

**Discussion by poojitharamachandra** [pymupdf/PyMuPDF#2735] [Q&A] [answered]
How to extract the table names in addition to table contents? Currently only header, columns, data inside table can be extracted.

**Comment by JorjMcKie** [comment]
I think you are referring to captions. Sorry, this is something highly dependent on the special making of each document. The caption may be missing altogether, or be above or below a table.
There especially is no **_technical_** connection between a table and its caption.
PDF does not even **_know_** that there is a table at all!

But you are the one who knows the document(s) that you are processing. So you can add code that extracts text in the table's neighbourhood, analyze it (whether it qualifies as a caption).

**Comment by poojitharamachandra** [comment]
thanks, any suggestion on how to identify the text in table neighbourhood(right before/after the table)?

**Comment by JorjMcKie** [comment, marked as answer]
> thanks, any suggestion on how to identify the text in table neighbourhood(right before/after the table)?

Not really failsafe ones. Maybe you try rectangle extractions above and below the table bbox.
Like if `tbbox` is the table bbox create a rectangle above for some height `h` like this: `ubbox = fitz.Rect(tbbox.x0, tbbox.y0 - h, tbbox.x1, tbbox.y0)`. Then extract the text in there: `caption = page.get_textbox(ubbox)` and see how far this takes you.
